numpy/core/src/umath/override.c

```diff
@@ -511,21 +509,18 @@ PyUFunc_CheckOverride(PyUFuncObject *ufunc, char *method,
 
         /* Choose an overriding argument */
         for (i = 0; i < num_override_args; i++) {
-            obj = with_override[i];
-            if (obj == NULL) {
+            override_obj = with_override[i];
+            if (override_obj == NULL) {
                 continue;
             }
 
-            /* Get the first instance of an overriding arg.*/
-            override_obj = obj;
-
             /* Check for sub-types to the right of obj. */
             for (j = i + 1; j < num_override_args; j++) {
-                other_obj = with_override[j];
+                PyObject *other_obj = with_override[j];
                 if (other_obj != NULL &&
-                    PyObject_Type(other_obj) != PyObject_Type(obj) &&
+                    Py_TYPE(other_obj) != Py_TYPE(override_obj) &&
```

Comment:
A nit: There's no point doing this extra check - it's just the first line of `PyObject_IsInstance` anyway.

Not going to make you change that in this PR though

Reply:
Surely, the first line is for `object == other` in `isinstance`, while here we do not want to defer to other if the types are the same. But I think you are right that the check can be removed, since we now explicitly remove instances of the same type...

Reply:
Nevermind, what I said above was wrong - it's not an optimization, it's a check for a strict subclass